thirdparty/CMakeLists.txt

```diff
@@ -3,76 +3,52 @@ include(FetchContent)
 ###################################################################################################
 # - spdlog ----------------------------------------------------------------------------------------
 
-FetchContent_Declare(
-    spdlog
-    GIT_REPOSITORY https://github.com/gabime/spdlog.git
-    GIT_TAG        v1.7.0
-    GIT_SHALLOW    true
-)
-
-FetchContent_GetProperties(spdlog)
-if(NOT spdlog_POPULATED)
-  FetchContent_Populate(spdlog)
-  add_subdirectory(${spdlog_SOURCE_DIR} ${spdlog_BINARY_DIR} EXCLUDE_FROM_ALL)
-endif()
+CPMAddPackage(
```

Comment:
Does CPM have the equivalent of GIT_SHALLOW (doesn't fetch the entire git history, which can be quite large)?

Reply:
Yes, I believe I can add back `GIT_SHALLOW` and it'll work as intended. CPM does automatically use shallow under certain circumstances:

| * Automatic shallow clone: if a version tag (e.g. v2.2.0) is provided and CPM_SOURCE_CACHE is used, CPM.cmake will perform a shallow clone of the dependency, which should be significantly faster while using less storage than a full clone.

But since `CPM_SOURCE_CACHE` isn't set (though it could be) by default this won't apply. I'll be happy to add back `GIT_SHALLOW` if that's preferred (I agree that it helps keeping existing behavior).

I'm not quite sure what motivates CPMs behavior. My guess is that it's to support offline work -- ie., if one sets `CPM_SOURCE_CACHE`, every dependency (say, in all of rapids) will be cloned (fully) only once, and then will stick around, and when it's used in a project like rmm or cudf, it'll just do a shallow clone into that project.